src/front/spv/mod.rs

```diff
@@ -4977,27 +4963,41 @@ impl<I: Iterator<Item = u32>> Frontend<I> {
         self.switch(ModuleState::Type, inst.op)?;
         inst.expect_at_least(3)?;
         let type_id = self.next()?;
+        let id = self.next()?;
+
         let type_lookup = self.lookup_type.lookup(type_id)?;
         let ty = type_lookup.handle;
-        let id = self.next()?;
 
         let mut components = Vec::with_capacity(inst.wc as usize - 3);
         for _ in 0..components.capacity() {
+            let start = self.data_offset;
             let component_id = self.next()?;
+            let span = self.span_from_with_op(start);
             let constant = self.lookup_constant.lookup(component_id)?;
-            components.push(constant.handle);
+            let expr = module
+                .const_expressions
+                .append(crate::Expression::Constant(constant.handle), span);
+            components.push(expr);
         }
 
+        let decor = self.future_decor.remove(&id).unwrap_or_default();
```

Comment:
I don't think this change will make a big difference since `.remove()` will already return an `Option<Decoration>` (there will already be enough stack space for the `Decoration`).

> it seems inefficient to whip up a whole default-valued `Decoration` just to get an optional name.

Is the concern about initializing all fields of the struct and then only using 2 of them?

Reply:
I will rethink about it when I continue the work on override expressions since I'll have to touch this code again.